### PR TITLE
Mesh party

### PR DESCRIPTION
**Pull request by cchriste**
Lots of additions, fixes, and cleanup for our upcoming release.

### CODE REVIEW

**Comment by cchriste** [comment]
@akenmorris The latest push to this branch is completely up to date with alpha and passes all unit tests (every executable ending with Test), so as long as everything is green it's ready to merge to alpha and should also fix the errors there.

**Comment by cchriste** [comment]
> so as long as everything is green it's ready to merge to alpha and should also fix the errors there. 

@medakk can you please take a look at the failing invert normals test on Linux?

The build itself is failing in Windows. I added eigen to target link libraries in hopes it will find the missing symbols.

**Comment by medakk** [comment]
@cchriste Should be fixed now

**Review by archanasri — APPROVED** [review]

**Comment by cchriste** [comment]
Windows build succeeded, two errors in testing:
1) `[  FAILED  ] OptimizeTests.embedded_python_test`
more than this, but says `Original error was: No module named 'numpy.core._multiarray_umath'`  
I thought we had this exact error before and that it had been resolved. @medakk @akenmorris ?

2) All the python tests now fail due to not being able to find the module. The pythonTests.setup test should set this up, and I just created an issue (#956) to do this the right way, but in the meantime is there something obvious missing, maybe path or pythonpath is no longer correct? I know we changed the default built type on Windows recently and I'm not sure we've had a working build for longer than that. 

Here's the output:
```
[==========] Running 71 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 71 tests from pythonTests
[ RUN      ] pythonTests.setup
path: D:/a/ShapeWorks/build\bin\Release;D:/a/ShapeWorks/build\bin\RelWithDebInfo;D:/a/ShapeWorks/build\bin\Debug;D:/a/ShapeWorks/build/install\bin\Release;D:/a/ShapeWorks/build/install\bin\RelWithDebInfo;D:/a/ShapeWorks/build/install\bin\Debug;C:\Miniconda3\envs\shapeworks;C:\Miniconda3\envs\shapeworks\Library\mingw-w64\bin;C:\Miniconda3\envs\shapeworks\Library\usr\bin;C:\Miniconda3\envs\shapeworks\Library\bin;C:\Miniconda3\envs\shapeworks\Scripts;C:\Miniconda3\envs\shapeworks\bin;C:\Miniconda3\Scripts\condabin;C:\Users\runneradmin\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\local\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Users\runneradmin\bin;C:\Miniconda3\condabin;D:\a\ShapeWorks\Qt\5.9.9\msvc2017_64\bin;C:\Users\runneradmin\AppData\Roaming\Python\Python39\Scripts;C:\hostedtoolcache\windows\Python\3.9.1\x64\Scripts;C:\hostedtoolcache\windows\Python\3.9.1\x64;C:\Users\runneradmin\.dotnet\tools;C:\Program Files\MongoDB\Server\4.4\bin;C:\aliyun-cli;C:\vcpkg;C:\cf-cli;C:\Program Files (x86)\NSIS;C:\Program Files\Mercurial;C:\hostedtoolcache\windows\stack\2.5.1\x64;C:\ProgramData\chocolatey\lib\ghc.8.10.3\tools\ghc-8.10.3\bin;C:\Program Files\dotnet;C:\mysql-5.7.21-winx64\bin;C:\Program Files\R\R-4.0.3\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\Program Files (x86)\sbt\bin;C:\Rust\.cargo\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\usr\bin;C:\Program Files (x86)\pipx_bin;C:\hostedtoolcache\windows\go\1.14.14\x64\bin;C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;C:\hostedtoolcache\windows\Python\3.7.9\x64;C:\hostedtoolcache\windows\Ruby\2.5.8\x64\bin;C:\Program Files\Java\jdk8u282-b08\bin;C:\npm\prefix;C:\Program Files\Microsoft SDKs\Azure\Azure Dev Spaces CLI;C:\Program Files\Microsoft SDKs\Azure\Azure Dev Spaces CLI;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0;C:\windows\System32\OpenSSH;C:\ProgramData\Chocolatey\bin;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Docker;C:\Program Files\PowerShell\7;C:\Program Files\dotnet;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\nodejs;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.6.3\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\OpenSSL\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\Program Files\TortoiseSVN\bin;C:\SeleniumWebDrivers\ChromeDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\Program Files\CMake\bin;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\bin;C:\Program Files (x86)\Microsoft BizTalk Server;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Git\usr\bin\vendor_perl;C:\Program Files\Git\usr\bin\core_perl

pythonpath: D:/a/ShapeWorks/build\bin\Release;D:/a/ShapeWorks/build\bin\RelWithDebInfo;D:/a/ShapeWorks/build\bin\Debug;D:/a/ShapeWorks/build/install\bin;D:/a/ShapeWorks/build/install\lib;

python.exe found in: C:\Miniconda3\envs\shapeworks
test environment setup complete!

[       OK ] pythonTests.setup (1 ms)
[ RUN      ] pythonTests.infoTest
Traceback (most recent call last):
  File "info.py", line 3, in <module>
    import shapeworks as sw

ModuleNotFoundError: No module named 'shapeworks'
2021-02-10T21:45:11.5172475Z 
```

**Comment by akenmorris** [comment]
> Original error was: No module named 'numpy.core._multiarray_umath' I thought we had this exact error before and that it had been resolved. @medakk @akenmorris ?

Not ringing any bells for me.

**Comment by akenmorris** [comment]
> I know we changed the default built type on Windows recently and I'm not sure we've had a working build for longer than that.

Windows build has been RelWithDebInfo since before the 5.5 release.

**Comment by akenmorris** [comment]
Didn't this PR successfully run all of the python tests?

https://github.com/SCIInstitute/ShapeWorks/runs/1850973262?check_suite_focus=true

This PR seems to have been set up against alpha, so github will have merged it with alpha prior to running.  What's changed since then?   (Maybe something on the runner?)

File: Testing/PythonTests/padfail.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-from shapeworks import *
+import shapeworks as sw
 
 def padfailTest1():
-  img = Image(os.environ["DATA"] + "/1x2x2.nrrd")
+  img = sw.Image(os.environ["DATA"] + "/1x2x2.nrrd")
   img.pad()
 
-  compareImg = Image(os.environ["DATA"] + "/padfail.nrrd")
+  compareImg = sw.Imageage(os.environ["DATA"] + "/padfail.nrrd")
```
Comment (cchriste) [inline Testing/PythonTests/padfail.py:9]: This fails for the wrong reasons @archanasri

**Comment by cchriste** [comment]
> This PR seems to have been set up against alpha, so github will have merged it with alpha prior to running. What's changed since then? (Maybe something on the runner?)

Nothing in the setup. I am suspicious of GitHub actions on this one, but I don't see anything obvious in the output. Does Windows use PATH or PYTHONPATH to import a module? Is it possible the version of python being run is 3.7  but the version compiled with is 3.9 (I know 3.9 is in there somewhere, but I don't know why).

**Comment by cchriste** [comment]
> > This PR seems to have been set up against alpha, so github will have merged it with alpha prior to running. What's changed since then? (Maybe something on the runner?)
> 
> Nothing in the setup. I am suspicious of GitHub actions on this one, but I don't see anything obvious in the output. Does Windows use PATH or PYTHONPATH to import a module? Is it possible the version of python being run is 3.7 but the version compiled with is 3.9 (I know 3.9 is in there somewhere, but I don't know why).

I just carefully checked (i.e., with diff) the output of the shapeworks pythonTests.setup and the only tiny difference is `C:\ProgramData\kind` comes later in the list on the pull request that succeeded. Both seem to have Python 3.9.1 in their path ahead of Python 3.7.9.

**Comment by cchriste** [comment]
> I just carefully checked (i.e., with diff) the output of the shapeworks pythonTests.setup and the only tiny difference is `C:\ProgramData\kind` comes later in the list on the pull request that succeeded. Both seem to have Python 3.9.1 in their path ahead of Python 3.7.9.

AH! The one that works builds this:
`shapeworks_py.vcxproj -> D:\a\ShapeWorks\build\bin\RelWithDebInfo\shapeworks.cp37-win_amd64.pyd`

The new one that fails builds this:
`shapeworks_py.vcxproj -> D:\a\ShapeWorks\build\bin\RelWithDebInfo\shapeworks.cp39-win_amd64.pyd`

File: Testing/PythonTests/padfail.py (hunk 1 above)
Comment (archanasri) [reply Testing/PythonTests/padfail.py:9]: Looks like it is fixed.

**Comment by cchriste** [comment]
Can we merge this? @akenmorris @archanasri

**Comment by archanasri** [comment]
> Can we merge this? @akenmorris @archanasri

Yes!

**Review by akenmorris — APPROVED** [review]

**Pull request merged**